tests/base.py

```diff
@@ -6,17 +8,58 @@
 from mod_regression.models import Category, RegressionTestOutput, RegressionTest, \
                                     regressionTestLinkTable, InputType, OutputType
 from mod_sample.models import Sample
-from collections import namedtuple
+from mod_auth.models import User, Role
 from unittest import mock
 from flask import g
 
 
+def generate_keys():
+    from utility import ROOT_DIR
+    secret_csrf_path = "{path}secret_csrf".format(path=os.path.join(ROOT_DIR, ""))
+    secret_key_path = "{path}secret_key".format(path=os.path.join(ROOT_DIR, ""))
+    if not os.path.exists(secret_csrf_path):
+        secret_csrf_cmd = "head -c 24 /dev/urandom > {path}".format(path=secret_csrf_path)
+        os.system(secret_csrf_cmd)
+    if not os.path.exists(secret_key_path):
+        secret_key_cmd = "head -c 24 /dev/urandom > {path}".format(path=secret_key_path)
+        os.system(secret_key_cmd)
+
+    return {'secret_csrf_path': secret_csrf_path, 'secret_key_path': secret_key_path}
+
+
 def load_config(file):
-    return {'Testing': True, 'DATABASE_URI': 'sqlite:///:memory:',
-            'WTF_CSRF_ENABLED': False, 'SQLALCHEMY_POOL_SIZE': 1,
-            'GITHUB_DEPLOY_KEY': 'test_deploy', 'GITHUB_CI_KEY': 'test_ci',
-            'GITHUB_TOKEN': '', 'GITHUB_BOT': '',
-            'GITHUB_OWNER': 'test_owner', 'GITHUB_REPOSITORY': 'test_repo'}
+    key_paths = generate_keys()
+    secret_key_file = open(key_paths['secret_key_path'], 'rb')
+    secret_key = secret_key_file.read()
+    secret_key_file.close()
+    secret_csrf_file = open(key_paths['secret_csrf_path'], 'rb')
+    secret_csrf = secret_csrf_file.read()
+    secret_csrf_file.close()
+
+    return {'Testing': True,
+            'DATABASE_URI': 'sqlite:///:memory:',
+            'WTF_CSRF_ENABLED': False,
+            'SQLALCHEMY_POOL_SIZE': 1,
+            'GITHUB_DEPLOY_KEY': 'test_deploy',
+            'GITHUB_CI_KEY': 'test_ci',
+            'GITHUB_TOKEN': '',
+            'GITHUB_BOT': '',
+            'GITHUB_OWNER': 'test_owner',
+            'GITHUB_REPOSITORY': 'test_repo',
+            'HMAC_KEY': 'test_key',
+            'MIN_PWD_LEN': '10',
+            'MAX_PWD_LEN': '500',
+            'SECRET_KEY': secret_key,
+            'CSRF_SESSION_KEY': secret_csrf
+            }
+
+
+class SignUpInformation:
```

Comment:
I don't really like creating a class just to store static fields in. Either use plain constants, or make a namedtuple of it.

Reply:
I added class so that in future maybe it could be extended with better ways (e.g. returning valid user info with `SignUpInfo.valid_user()`, or exiting user `SignUpInfo.existing_user()`. But maybe that can be added later when needed. Removed for now.